### DISCUSSION TITLE
Has the ability to bring up a DAGs previous version view + logs been removed between v2.3.3 and v2.5.1?

### DISCUSSION THREAD

**Discussion by SbtrktCtrl** [apache/airflow#33817] [Q&A] [answered]
We have recently upgraded from v2.3.3 to 2.5.1 and the link to previous DAG versions from the DAG Runs list no longer works. 

In version 2.3.3, when viewing Browse -> Dag Runs list, if a Dag had been deployed, run and then re-deployed (i.e. code change and new ID, so <dag name>-ID1 to <dag name 2>-ID2) it was a) visible on the dag runs list, and b) when you clicked on it you could still view the DAG info in Grid/Graph view w/ logs etc. e.g. https://airflow.apache.org/docs/apache-airflow/stable/ui.html#grid-view

In v2.5.1 we can still see a previous DAG version in the DAG Runs list, however when you click on the DAG itself we get an error - "<dag_name> appears to be missing".

Note: I am not expecting to see the previous version in the main DAGs view: https://airflow.apache.org/docs/apache-airflow/stable/ui.html#dags-view, only in the DAG runs list. 

I am trying to determine if this functionality has been removed between v2.3.3 and 2.5.1? or there is potentially a config or other error.

Troubleshooting:
- reviewed release notes
- reviewed logs (The older dag version logs are still present on the server, there are no obvious errors in the logs.)
- ruled out permissions issues.

**Comment by potiuk** [comment]
Not sure @bbovenzi @pierrejeambrun  - doe sit ring a bell as one of possible changes to the UI ?

**Comment by pierrejeambrun** [comment, marked as answer]
That doesn't ring a bell on my side.

Nevertheless grid view doesn't seem accessible for deleted Dags, i.e renamed into something new. Indeed those runs still appear on the `DagRun Browse list` and we have an error when clicking on it from there. (redirecting to grid+graph doesn't work on main).

See bellow, (`dynamic_task_group` was renamed into `dynamic_task_group`). Error appears when clicking on one of the dagruns from `dynamic_task_group`, that do not exist anymore. We should fix that

![image](https://github.com/apache/airflow/assets/14861206/0532b941-b1fb-4121-9f9e-f4469a07e65e)
![image](https://github.com/apache/airflow/assets/14861206/3050e02d-4bac-4fc5-b8f3-e3c8a11f10a8)

**Comment by pierrejeambrun** [comment]
Issue created here:
https://github.com/apache/airflow/issues/34058
Closing in favor of the issue tracker, thank you for reporting, feel free to reopen if needed.

**Reply by SbtrktCtrl** [reply]
Legends! thank you.
